### PR TITLE
expect events of type EmailEvent

### PR DESCRIPTION
**Pull request by edevil**
Support EmailEvent type when tailing logs of a worker.

What this PR solves / how to test:

Author has included the following, where applicable:

- [X] Tests
- [X] Changeset

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

### CODE REVIEW

File: .changeset/sixty-books-share.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": minor
+---
+
+support EmailEvent event type
```
Comment (petebacondarwin) [inline .changeset/sixty-books-share.md:5]:
```suggestion
feat: support EmailEvent event type in `wrangler tail`.
```

Comment (petebacondarwin) [reply .changeset/sixty-books-share.md:5]: See https://github.com/cloudflare/wrangler2/blob/main/CONTRIBUTING.md#changeset-message-format

**Review by petebacondarwin — APPROVED** [review]
Love this PR! Tests, changeset, helper functions. It's got the lot.
High quality contribution @edevil - thanks

File: .changeset/sixty-books-share.md (hunk 1 above)
Comment (edevil) [reply .changeset/sixty-books-share.md]: Done, thanks.

**Review by JacobMGEvans — APPROVED** [review]
Strongly agree with Pete, this is a great and concise PR

**Comment by petebacondarwin** [comment]
One thing I didn't do - sorry. Is to actually create a Worker that receives emails and show that tail works correctly here.
@edevil - did you prove this locally?

**Comment by edevil** [comment]
@petebacondarwin I didn't because the edgeworker change to actually send these events is still being deployed as we speak. As soon as it is RoW I'll test.

**Comment by edevil** [comment]
I just tested this on EW version 800 and it's working as intended.

**Pull request merged**